addons/mass_mailing/static/src/builder/plugins/social_media_option_plugin.js

```diff
@@ -0,0 +1,429 @@
+        }
+        return linkToEdit.getAttribute("href");
+    }
+}
+
+class ChangeCompanyAction extends BuilderAction {
+    static id = "changeCompany";
+    static dependencies = ["massMailingSocialMediaOptionPlugin"];
+    async apply({ editingElement, params: company }) {
+        if (this.isApplied(...arguments)) {
+            return;
+        }
+        editingElement.querySelector(".o_social_snippet_dropped_placeholder")?.remove();
+        const baseMedias =
+            await this.dependencies.massMailingSocialMediaOptionPlugin.fetchRecordedSocialMedia(
+                company.id
+            );
+        const allLinks = [];
+        for (const [platform, href] of Object.entries(baseMedias)) {
+            const newLinkElement =
+                this.dependencies.massMailingSocialMediaOptionPlugin.renderSocialMediaLink(
+                    platform,
+                    href
+                );
+            allLinks.push(newLinkElement);
+        }
+        if (
+            !allLinks.length &&
+            !editingElement.querySelector(".o_social_snippet_empty_placeholder")
+        ) {
+            const placeholder =
+                this.dependencies.massMailingSocialMediaOptionPlugin.renderPlaceholderEl();
+            allLinks.push(placeholder);
+        }
+        editingElement.querySelectorAll("a[data-platform]").forEach((element) => {
+            element.remove();
+        });
+        editingElement.firstElementChild.after(...allLinks);
+        editingElement.dataset.companyId = company.id;
+    }
+    isApplied({ editingElement, params: company }) {
+        return parseInt(editingElement.dataset.companyId) === company.id;
+    }
+}
+
+class ToggleSocialMediaLinkAction extends BuilderAction {
+    static id = "toggleSocialMediaLink";
+    static dependencies = ["builderActions", "massMailingSocialMediaOptionPlugin"];
+    setup() {
+        this.classAction = this.dependencies.builderActions.getAction("classAction");
+        this.styleAction = this.dependencies.builderActions.getAction("styleAction");
+    }
+
+    apply({ editingElement, params: { platform, getIndex, href } }) {
+        const editingLink = editingElement.querySelector(`[data-platform="${platform}"]`);
+        if (href) {
+            this.dependencies.massMailingSocialMediaOptionPlugin.addMedia(platform, href);
+        }
+        href = this.dependencies.massMailingSocialMediaOptionPlugin.getMedias()[platform];
+        const newLinkElement =
+            this.dependencies.massMailingSocialMediaOptionPlugin.renderSocialMediaLink(
+                platform,
+                href
+            );
+        if (editingLink) {
+            editingLink.remove();
+        } else if (getIndex() === 0) {
+            const firstLink = editingElement.querySelector("[data-platform]:first-of-type");
+            firstLink ? firstLink.before(newLinkElement) : editingElement.append(newLinkElement);
```

Comment:
same remark as other comment, and also valid for some `lastElementChild.after` ->
shouldn't we insert before the `o_social_snippet_empty_placeholder` instead of append?

to investigate if this is correct